### PR TITLE
types: fix some new types for handlers & helpers & structures

### PR DESCRIPTION
**Pull request by itohatweb**

### CODE REVIEW

File: src/handlers/commands/APPLICATION_COMMAND_UPDATE.ts [hunk 1]
```diff
@@ -9,9 +13,9 @@ export function handleApplicationCommandUpdate(data: DiscordGatewayPayload) {
...
-  eventHandlers.applicationCommandUpdate?.({
-    ...rest,
-    guildId,
-    applicationId,
-  });
+  eventHandlers.applicationCommandUpdate?.(
+    snakeKeysToCamelCase<ApplicationCommandCreateUpdateDelete>(
+      data.d as DiscordApplicationCommandCreateUpdateDelete,
+    ),
+  );
```
Comment (ayntee) [inline src/handlers/commands/APPLICATION_COMMAND_UPDATE.ts:20]:
```ts
const { ...rest } = ...
```
This should be removed entirely

File: src/handlers/emojis/GUILD_EMOJIS_UPDATE.ts [hunk 2]
```diff
@@ -11,7 +11,7 @@ export async function handleGuildEmojisUpdate(data: DiscordGatewayPayload) {
 
   const cachedEmojis = guild.emojis;
   guild.emojis = new Collection(
-    payload.emojis.map((emoji) => [emoji.id ?? emoji.name, emoji]),
+    payload.emojis.map((emoji) => [emoji.id!, emoji]),
```
Comment (ayntee) [inline src/handlers/emojis/GUILD_EMOJIS_UPDATE.ts:14]: Explain why this is done

Comment (itohatweb) [reply src/handlers/emojis/GUILD_EMOJIS_UPDATE.ts:14]: This event is only emitted when a custom emoji is updated which always has an id.

File: src/handlers/guilds/GUILD_UPDATE.ts [hunk 3]
```diff
@@ -30,6 +30,8 @@ export async function handleGuildUpdate(data: DiscordGatewayPayload) {
         if (Array.isArray(cachedValue) && Array.isArray(value)) {
           const different = (cachedValue.length !== value.length) ||
             cachedValue.find((val) => !value.includes(val)) ||
+            // TODO: check if this really works hehe
+            // @ts-ignore this works ts is wrong
```
Comment (ayntee) [inline src/handlers/guilds/GUILD_UPDATE.ts:34]: "this works ts is wrong," please refrain from adding these trivial comments, and add meaningful and descriptive reason as to why `// @ts-ignore` is added

**Review by Skillz4Killz — APPROVED** [review]

**Pull request merged**